students/Cracolici_Jon/lesson05/JonCracolici_Mailroom3.py

```diff
@@ -0,0 +1,180 @@
+    new_donor(newdonordict, dB)
+    note_gen(dB[newdonordict['donornamekey']])
+    return dB#, start_control(dB)
+def ud_control(dB):
+    """Function that calls functions to update a donor in the database, print letter to screen."""
+    updonordict = donor_input(dB)
+    if updonordict == False:
+        return False
+    elif updonordict['donornamekey'] not in dB.keys():
+        print("It appears that you are attempting t update a donor not already in our system.\n Perhaps you would like to select the 'New Donor' option instead?")
+        return False
+    #will add try except to check for donor actually being new in L5
+    update_donor(updonordict, dB)
+    note_gen(dB[updonordict['donornamekey']])
+    return dB#, start_control(dB)
+def donor_input(dB):
+    """This function collects user input about the donation."""
+    #create a set of quit criteria
+    quitset = set(['quit', 'q'])
+    donorname = input('Who is the donor?')
+    #Check if user wants to exit to previous menu
+    if donorname.lower() in quitset:
+        return False
+    newdonation = input('How much did they donate?')
+    #Check if user wants to exit to previous menu
+    if newdonation.lower() in quitset:
+        return False
+        
+    else:#this section creates the keys and values for the outputted dict
+        donornamekey = donorname.lower()
+        donornamekey = donornamekey.replace(' ', '_')
+        donor = donorname.title()
+        try:
+            donation = float(newdonation)
+        except ValueError:
+            print("Use numbers for the donation amount. NUMBERS!")
+            return False
+        newdondict = {'donornamekey': donornamekey, 'donor': donor, 'donation': donation}
+        return newdondict
+def print_letters(donordict):
+    """This function takes a string and makes a document with it."""
+    lettername = donordict['fname']+'.txt'
+    outfile = open(lettername, 'w')
+    outfile.write(note_gen(donordict, dest = 'f'))
+    outfile.close()
+def note_gen(persondict, dest = 's'):
+    """Creates a thank you note. Includes a required positional argument of the intended recipient's info,
+    and a kwarg "dest" that defaults to printing to screen but may be set to generate a file."""
+    message = "Dear {donor}, \nThank you for your generous donation of ${lastdonation:.02f}. Please rest assured that we will use at least \n95% of your contribution to feed the homeless to wolves. We could not do this work without you. \nSincerely, \nThe Billionaires' Club".format(**persondict)
+    if dest == 's':
+        print(message)
+    elif dest == 'f':
+        return message
+def quit_program(db):
+    print("Goodbye dear User!")
+    return False
+    #sys.exit()
+def quit_saty(db):
+    print("Bye bye!")
+    return False
+def main():
+    db = initialize_database()
+    while True:
+        x = start_control(db)
+        if x == False:
+            break
+def main_2(db):
```

Comment:
per pep8 make sure to have 2 lines space between module functions; code is a bit difficult to read with no spaces.

Reply:
ok will do